src/sql/src/connectors.rs

```diff
@@ -18,12 +24,51 @@ use crate::catalog::SessionCatalog;
 use crate::normalize;
 use crate::plan::{PlanError, StatementContext};
 
+lazy_static! {
```

Comment:
@guswynn Suggested this should be a [once_cell](https://docs.rs/once_cell/latest/once_cell/) but since I know how to use `lazy_static!` I decided to start with this and then its an easy migration to `once_cell` is that is the right choice in context.

Reply:
```
static SECRETS: once_cell::sync::Lazy<Arc<Box<dyn SecretsReader>>> = once_cell::sync::Lazy::new(|| default_secrets_reader());
```

we can fix it later but its not too bad!

Reply:
The existence of `default_secrets_reader` makes me uncomfortable. I do not understand the RUST reasons for why we need the `lazy_static` and `default_secrets_reader`

Reply:
The `lazy_static` is so we only pay the initialization cost once, small though it may be. 
The reason for the default reader is that right now only one narrow slice of queries require reading secrets during planning so plumbing it in from the coordinator to be used in this one place while I also had so much other change in motion seemed like a bad tradeoff for this stage of the code evolution. There are also design routes forward which eliminate the need for the planner to read secrets and would instead push that out to where the secrets are being used in clients but again they require more substantial changes.
Because of how intertwined these authentication credentials are with parsing, purification, planning and post render runtime this was the most reasonable compromise I could come up with as an interim step.

The `SecretsReader` here is a trait which is why it's `dyn` since despite there only being the 1 struct implementing it I think its worthwhile to not tie ourselves to forcing this one struct to handle every future use-case.

Reply:
> The reason for the default reader is that right now only one narrow slice of queries require reading secrets during planning so plumbing it in from the coordinator to be used in this one place while I also had so much other change in motion seemed like a bad tradeoff for this stage of the code evolution.

Trying to keep this PR small is exactly right, but hardcoding paths and reading environment variables in the SQL planner is a non-starter, sorry! The trait looks great, but the implementation 100% needs to be properly plumbed from the main CLI driver. 

Why don't we separate out the `SecretsReader` and its plumbing into a PR to merge first? That is a very small, targeted PR that should be able to land without too much fuss. It's totally okay that the `SecretsReader` won't be hooked up to anything to start.